Add ability for user to specify excluded networks when generating a FLIX

bthaile opened the pull request:
Closes: https://github.com/onflow/flixkit-go/issues/58

## Description

Address issue where users wants to exclude specific network(s) when generating a FLIX. Added new `exclude-networks` command line flag for FLIX command.



______

For contributor use:

- [ ] Targeted PR against `master` branch
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work
- [ ] Code follows the [standards mentioned here](https://github.com/onflow/flow-cli/blob/master/CONTRIBUTING.md#styleguides)
- [x] Updated relevant documentation
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Added appropriate labels

chasefleming reviewed (commented):
Don't different networks have different versions of Cadence? Can you elaborate on how this works please?

bthaile commented on internal/super/flix.go:
```diff
@@ -49,6 +49,7 @@ type flixFlags struct {
 	GasLimit    uint64   `default:"1000" flag:"gas-limit" info:"transaction gas limit"`
 	PreFill     string   `default:"" flag:"pre-fill" info:"template path to pre fill the FLIX"`
 	Lang        string   `default:"js" flag:"lang" info:"language to generate the template for"`
+	Networks    []string `default:"" flag:"networks" info:"networks to generate the template for"`
```
geesh, good catch. 
User can add 0 or many networks. relied on copilot too much on that one.
updated my PR. Thanks

ianthpun replied:
@bthaile are you good with changing the name to something other thna `networks`? Reason is that we already have a pre-existing `network` flag, and this just might add more confusion

chasefleming replied:
Yeah because there is a global flag named network, I think it needs to work with that. `networks` is too confusing to someone familiar with the global flags.

sisyphusSmiling replied:
What about `--exclude-networks`? Default is to run against all networks and anything listed after flag is excluded

bthaile replied:
Changed it to `Targets` I think that is fine

chasefleming replied:
Saying "targets" is for "target networks" still holds about the same amount of confusion in my opinion and starts to introduce new patterns that may start to stink into other parts of code. I'd go with what @sisyphusSmiling said, `--exlcude-networks`, as it has the potential to be more applicable across other use cases if something similar is ever needed.

bthaile commented:
> Don't different networks have different versions of Cadence? Can you elaborate on how this works please?

Yes, different networks need different version of cadence. This version is cadence v1 compatible. so user will have to pass in the networks that are cadence v1 compatible. They can use old version of flow-cli to get pre cadence v1 compatibility.

Eventually there will only be cadence v1.

bthaile commented:
> Built and tried it out locally - seems to be generating FLIX just fine against previewnet!

can you also execute the flix?

sisyphusSmiling commented:
> can you also execute the flix?

I wouldn't be able to execute the particular flix, but that's an edge case where the Previewnet version isn't the one that I need for me to be able to execute the template's transaction code. I'll try once EVM on Previewnet is updated and let you know, hopefully by EOD today.

chasefleming commented:
Can you also update the pr title? Something that will work for the release notes please?

bthaile commented:
> Can you also update the pr title? Something that will work for the release notes please?

Good point, I changed the flag to Targets and updated the title and description to be more release notes friendly.

bthaile commented:
> Can you also update the pr title? Something that will work for the release notes please?

Made all the updates, does the title and description make sense?
